File: Beginner Solutions/F Adding Fractions with Nancy/AddingFractionsWithNancy.java
public class AddingFractionsWithNancy{
   public static void main(String[]args){
   
   }
   public static String addFrac(int num1, int denom1, int num2, int denom2){
      if(denom1 != denom2){
         int lcm = 0;
         for(int i=2; i<=denom1*denom2; i++){
            if(i % denom1 == 0 && i % denom2 == 0) lcm = i;
         } 
         return (num1*(lcm/denom1) + num2*(lcm/denom2)) + "/" + lcm;
      }
      else 
         return (num1 + num2) + "/" + denom1;  
   }
}

File: Beginner Solutions/F Adding Fractions with Nancy/AddingFractionsWithNancy.py
def addFrac(n1,d1,n2,d2):
    finald = d1*d2
    finaln1 = n1 * d2
    finaln2 = n2 * d1
    finaln = finaln1+finaln2
    return str(finaln)+"/"+str(finald)
num1 = int(input())
denom1= int(input())
num2 = int(input())
denom2 = int(input())
print(addFrac(num1,denom1,num2,denom2))